### PR TITLE
TypeScript config

### PR DESCRIPTION
**Pull request by FranciscoMCG**
Added TypeScript config bor both Front End and Back End.

### CODE REVIEW

File: api/.eslintrc.json [hunk 1]
```diff
@@ -1,17 +1,27 @@
...
+    "Atomics": "readonly",
+    "SharedArrayBuffer": "readonly"
+  },
+  "parser": "@typescript-eslint/parser",
+  "extends": ["plugin:@typescript-eslint/recommended"],
+  "parserOptions": {
+    "ecmaVersion": 2018
+  },
+  "rules": {
+    "quotes": ["error", "double"],
```
Comment (TdfGomes) [inline api/.eslintrc.json]: Cloud this be an warning instead an error ?
Don't understand why this is an error, is something that TS required ?

Comment (TdfGomes) [inline api/.eslintrc.json]: Cloud this be an warning instead an error ?
Don't understand why this is an error, is something that TS required ?

Comment (FranciscoMCG) [reply api/.eslintrc.json]: According to the docs it needs to be "error" but... I'm not even a fan of double quotes. I prefer single, and you?  

We can also use prettier to enforce this.

Comment (TdfGomes) [reply api/.eslintrc.json]: That's a good idea. Could you add a `.prittierrc` file to this PR ?

Comment (FranciscoMCG) [reply api/.eslintrc.json]: sure no problem. And I'd make this single quotes.

**Pull request merged**